test/client.js

```diff
@@ -64,4 +65,62 @@ describe('Client', function () {
       });
     });
   });
+  describe('fixtures: icy-server-https-response', function () {
+      it('should not get a Parse Error with https server', function (done) {
+        var options = {
+            key: fs.readFileSync('test/fixtures/key.pem'),
+            cert: fs.readFileSync('test/fixtures/cert.pem')
+        };
+        //Self signed certificate
+        process.env.NODE_TLS_REJECT_UNAUTHORIZED = "0";
```

Comment:
Change this to `rejectUnauthorized: false` in the `options` object. No need to touch the process env.